docs/architecture/adr-023-protobuf-naming.md

```diff
@@ -99,13 +99,67 @@ the best choices for the future.
 
 ### Versioning
 
-#### Don't Allow Breaking Changes in Stable Packages
-
-Always use a breaking change detector such as [Buf](https://buf.build) to prevent
-breaking changes in stable (non-alpha or beta) packages. Breaking changes can
-break smart contracts/persistent scripts and generally provide a bad UX for
-clients. With protobuf, there should usually be ways to extend existing
-functionality instead of just breaking it.
+#### Guidelines on Stable Package Versions
+
+In general, schema evolution is the way to update protobuf schemas. That means that new fields,
+messages, and RPC methods are _added_ to existing schemas and old fields, messages and RPC methods
+are maintained as long as possible.
+
+Breaking things is often unacceptable in a blockchain scenario. For instance, immutable smart contracts
+may depend on certain data schemas on the host chain. If the host chain breaks those schemas, the smart
+contract may be irreparably broken. Even when things can be fixed (for instance in client software),
+this often comes at a high cost.
+
+Instead of breaking things, we should make every effort to evolve schemas rather than just breaking them.
+[Buf](https://buf.build) breaking change detection should be used on all stable (non-alpha or beta) packages
+to prevent such breakage.
+
+With that in mind, different stable versions (i.e. `v1` or `v2`) of a package should more or less be considered
+different packages and this should be last resort approach for upgrading protobuf schemas. Scenarios where creating
+a `v2` may make sense are:
+* we want to create a new module with similar functionality to an existing module and adding `v2` is the most natural
+way to do this. In that case, there are really just two different, but similar modules with different APIs.
+* we want to add a new revamped API for an existing module and it's just too cumbersome to add it to the existing package,
+so putting it in `v2` is cleaner for users. In this case, care should be made to not deprecate support for
+`v1` if it is actively used in immutable smart contracts.
+
+An alternative to creating `v2` packages is creating `v2` messages and RPC methods. For instance say we found a better API
+for sending coins, then we could add `MsgSendV2` to the existing bank package without needing to change `MsgSend`
+or create a `bank.v2` package.
+
+#### Guidelines on unstable (alpha and beta) package versions
+
+The following guidelines are recommended for marking packages as alpha or beta:
+* marking something as `alpha` or `beta` should be a last resort and just putting something in the
+stable package (i.e. `v1` or `v2`) should be preferred
+* a package *should* be marked as `alpha` *if and only if* there are active discussions to remove
+or significantly alter the package in the near future
+* a package *should* be marked as `beta` *if and only if* there is an active discussion to 
+significantly refactor/rework the functionality in the near future but not remove it
```

Comment:
As described here, these 2 seems pretty similar to me: I don't see clearly the difference between `significantly alter the package` and `significantly refactor/rework the functionality`. For example, I sometimes here people say "x/bank" should be revamped, should it go as alpha or beta?

We could offer the same guarantees as [Kubernetes API](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning):
- alpha: "The API may change in incompatible ways in a later software release without notice."
- beta: "The schema and/or semantics of objects may change in incompatible ways in a subsequent beta or stable release. When this happens, we will provide instructions for migrating to the next version."

Reply:
What you describe is how I read the current version:

"a package *should* be marked as `alpha` *if and only if* there are active discussions to **remove or** significantly alter the package in the near future"

Reply:
Is there an update you would suggest @amaurymartiny ? I based this off of what I read in Kubernetes btw.